### PR TITLE
[Router] Add useMatch for crafting stateful links

### PR DESCRIPTION
**Pull request by RobertBroersma**
It's _almost_ a direct copy from `@reach/router` or `react-router@6`, but using the `matchPath` function that was already in place!

This should fulfill my wishes from #360 to craft stateful links.

_NOTE: This changes the current implementation of `NavLink`! Instead of **replacing** the `className` with `activeClassName`, it now **joins** the `className` and the `activeClassName`_

### CODE REVIEW

File: packages/router/src/links.js [hunk 1]
```diff
@@ -16,8 +28,11 @@ const Link = forwardRef(({ to, ...rest }, ref) => (
 
 const NavLink = forwardRef(
   ({ to, className, activeClassName, ...rest }, ref) => {
-    const context = useContext(LocationContext)
-    const theClassName = to === context.pathname ? activeClassName : className
+    const matchInfo = useMatch(to)
+    const theClassName = matchInfo.match
+      ? `${className || ''} ${activeClassName}`
```
[the remark this answers is not in the document]
Comment (RobertBroersma) [reply packages/router/src/links.js:33]: Preventing leading space sounds like a good idea!

Not sure what you mean by I'm changing the behavior. Am I not keeping the `className` and adding the `activeClassName`?

[the remark this answers is not in the document]
Comment (gfpacheco) [reply packages/router/src/links.js:33]: Yes you are, and although that's what I think should be done, it's not how it currently works:

```
const theClassName = to === context.pathname ? activeClassName : className
```

[the remark this answers is not in the document]
Comment (RobertBroersma) [reply packages/router/src/links.js:33]: You are right! 

I tried the current way in v0.3.2 and couldn't get it to work. This is also what I mentioned in the issue I created. I haven't tried in v0.4.0 tho!

There's no documentation yet for `NavLink` to change the communication (cmiiw). I could update the commit or we can make sure to add it to the changelog?

**Comment by thedavidprice** [comment]
FYI: we’ve added a CI runner for Windows and changed PR settings to require checks to pass before merging. Pulling in changes from master should resolve current warnings.

File: packages/router/src/__tests__/links.test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import { toHaveClass, toHaveStyle } from '@testing-library/jest-dom/matchers'
+expect.extend({ toHaveClass, toHaveStyle })
+
+import { NavLink, useMatch, Link } from '../links'
+import { LocationProvider } from '../location'
+
+describe('<NavLink />', () => {
+  it('receives active class on the same path', () => {
+    const mockLocation = {
+      pathname: '/dunder-mifflin',
```
Comment (peterp) [inline packages/router/src/__tests__/links.test.js:12]: I hear `/michael-scott-paper-company` gives a 200 response.

**Comment by thedavidprice** [comment]
Hi all. Delayed question here -- did we ever get docs for this? (Or do we even need docs for this?)

Thanks again!

**Comment by RobertBroersma** [comment]
> Hi all. Delayed question here -- did we ever get docs for this? (Or do we even need docs for this?)
> 
> Thanks again!

We did not! I'll whip up some docs this weekend.

**Pull request merged**